src/ui/win32/bindings/MemoryViewerControlBinding.cpp

```diff
@@ -236,6 +239,68 @@ bool MemoryViewerControlBinding::HandleNavigation(UINT nChar)
     }
 }
 
+bool MemoryViewerControlBinding::HandleShortcut(UINT nChar)
+{
+    const bool bShiftHeld = (GetKeyState(VK_SHIFT) < 0);
+    const bool bControlHeld = (GetKeyState(VK_CONTROL) < 0);
+
+    switch (nChar)
+    {
+        case 'C':
+            if (bControlHeld)
+            {
+                const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+                const auto nValue = pEmulatorContext.ReadMemory(m_pViewModel.GetAddress(), m_pViewModel.GetSize());
+                std::wstring sValue = ra::data::MemSizeFormat(nValue, m_pViewModel.GetSize(), MemFormat::Hex);
+
+                ra::services::ServiceLocator::Get<ra::services::IClipboard>().SetText(ra::Widen(sValue));
+            }
+            return true;
+
+        case 'V':
+            if (bControlHeld and !m_pViewModel.IsReadOnly())
+            {
+                auto nAddress = m_pViewModel.GetAddress();
+                const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+                std::wstring sClipboardText = ra::services::ServiceLocator::Get<ra::services::IClipboard>().GetText();
+                const auto nNibblesForSize = ra::data::MemSizeBytes(m_pViewModel.GetSize());
+                std::vector<std::uint32_t> v_nValues;
+                std::wstring sError;
+
+                if (sClipboardText.empty())
+                    return false;
+
+                // Split the clipboard value into substrings matching the current size and check if they're valid hexadecimal values
+                for (size_t i = 0; i < sClipboardText.size(); i += nNibblesForSize * 2)
+                {
+                    std::wstring sSubString = sClipboardText.substr(i, nNibblesForSize * 2);
+                    unsigned int nValue;
+                    
+                    if (!ra::ParseHex(sSubString, 0xFFFFFFFF, nValue, sError))
+                    {
+                        ra::ui::viewmodels::MessageBoxViewModel::ShowWarningMessage(L"Paste value failed", sError);
+                        return false;
+                    }
+
+                    v_nValues.push_back(nValue);
+                }
+
+                for (auto nValue : v_nValues)
+                {
+                    //Single mode writes only the first value, multi mode (shift) writes every values
+                    pEmulatorContext.WriteMemory(nAddress, m_pViewModel.GetSize(), nValue);
+                    if (!bShiftHeld)
```

Comment:
Recommend moving the shift handling logic into the above loop. No reason to populate the entire array if you're only going to process a single item.